### PR TITLE
Support bulkhead group

### PR DESCRIPTION
**Pull request by Ferioney**
Related to [Spring Cloud OpenFeign Issue 514](https://github.com/spring-cloud/spring-cloud-openfeign/issues/514).
These changes required the latest version of Spring Cloud Commons.

### CODE REVIEW

**Review by ryanjbaxter — CHANGES_REQUESTED** [review]
Please open the PR against the 1.0.x branch

**Comment by Ferioney** [comment]
PR to the 1.0.x branch [104](https://github.com/spring-cloud/spring-cloud-circuitbreaker/pull/104/files)

File: spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JCircuitBreaker.java [hunk 1]
```diff
@@ -76,6 +79,25 @@ public Resilience4JCircuitBreaker(String id,
...
+		this.circuitBreakerConfig = circuitBreakerConfig;
+		this.registry = circuitBreakerRegistry;
+		this.timeLimiterRegistry = timeLimiterRegistry;
+		this.timeLimiterConfig = timeLimiterConfig;
+		this.executorService = executorService;
+		this.circuitBreakerCustomizer = circuitBreakerCustomizer;
+		this.bulkheadProvider = bulkheadProvider;
+	}
+
+	public Resilience4JCircuitBreaker(String id,
```
Comment (ryanjbaxter) [inline spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JCircuitBreaker.java:92]: Can we have the old constructor call this one?

File: spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JCircuitBreakerFactory.java [hunk 2]
```diff
@@ -108,6 +115,18 @@ public Resilience4JCircuitBreaker create(String id) {
 				Optional.ofNullable(circuitBreakerCustomizers.get(id)), bulkheadProvider);
 	}
 
+	@Override
+	public org.springframework.cloud.client.circuitbreaker.CircuitBreaker create(String id, String groupName) {
```
Comment (ryanjbaxter) [inline spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JCircuitBreakerFactory.java:119]: Could we find a way to reuse code between the two `create` methods?

**Review by ryanjbaxter — CHANGES_REQUESTED** [review]

**Pull request closed without merging**